### PR TITLE
ENH: allow mapping reads to single reference genome by map-reads

### LINKED ISSUE #131: ENH: allow mapping reads to individual genes
**Is your feature request related to a problem? Please describe.**
Currently, if I have a list of genes (or other DNA sequences) I am unable to map reads to them using any of the available actions - those only expect assembled contigs or MAGs as reference. If I want to estimate abundance of specific genes in my experiment could just map reads directly to them.

**Describe the solution you'd like**
Ideally we could reuse the existing `map-reads` action and just expand it by allowing a new reference semantic type. If that is not possible, we could create a new action `map-reads-to-genes` or something like that.

### PR DESCRIPTION
**Pull request by misialq**
# Description

Closes #131 

Testing note: please test with all the kinds of inputs to make sure that nothing broke for those.

# AI Disclosure

- [x] NO AI USED.
- [ ] AI USED.

### CODE REVIEW

**Comment by VinzentRisch** [comment]
Hey @misialq 

Sorry for the delay I forgot about this. I now ran all the possible inputs and it all runs without problems. I have to leave now and only have time to review the code tomorrow afternoon. If that is okay then I can do that. If you need it earlier then somebody else has to have a look. I'm very sorry for this. I had a bit of problems to get my docker container running again.

**Comment by misialq** [comment]
Hey @VinzentRisch, no worries, tomorrow is also fine. Thank you!

File: q2_assembly/bowtie2/mapping.py [hunk 1]
```diff
@@ -419,11 +421,14 @@ def _gather_feature_data(
         full_set (dict): Dictionary with read and index information per sample.
     """
     full_set = {}
+    index_prefix = os.path.basename(glob.glob(os.path.join(str(index), "*"))[0]).split(
+        ".", 1
+    )[0]
```
Comment (VinzentRisch) [inline q2_assembly/bowtie2/mapping.py]: This approach will fail for index names with a ".". So if someone would import an index this could potentially be a problem. Because the suffixes are always the same in the bowtie indexes you can parse the suffix and remove that from the string.

File: q2_assembly/bowtie2/tests/test_mapping.py [hunk 2]
```diff
@@ -148,24 +148,61 @@ def test_gather_sample_data_paired_missing_sample(self):
...
+                ),
+            )
+
+    def test_gather_feature_data_empty_index_raises(self):
+        """An empty index directory raises IndexError from the glob lookup."""
+        manifest = self.read_manifest_file("paired")
+        index = Bowtie2IndexDirFmt()  # no files added
+
+        with self.assertRaises(IndexError):
+            _gather_feature_data(index=index, reads_manifest=manifest, paired=True)
```
Comment (VinzentRisch) [inline q2_assembly/bowtie2/tests/test_mapping.py]: Why do we need this test? The Bowtie2IndexDirFmt() after validation cannot be empty. Or is that it is empty expected?

**Review by VinzentRisch — CHANGES_REQUESTED** [review]
Hey @misialq 
I have a suggestion and a question below :)

File: q2_assembly/bowtie2/tests/test_mapping.py (hunk 2 above)
Comment (misialq) [reply q2_assembly/bowtie2/tests/test_mapping.py]: Ooops, true, this is not possible - I removed the test, thanks!

File: q2_assembly/bowtie2/mapping.py (hunk 1 above)
Comment (misialq) [reply q2_assembly/bowtie2/mapping.py]: Excellent point. So I actually just discovered that the directory format has a helper method to do just that which greatly simplifies this code here.

Comment (VinzentRisch) [reply q2_assembly/bowtie2/mapping.py]: Ah yes that is even better!

**Review by VinzentRisch — APPROVED** [review]
This looks good! Ready for merge :)
